persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/AbstractPersistentEntityRegistry.scala

```diff
@@ -67,6 +67,10 @@ abstract class AbstractPersistentEntityRegistry(system: ActorSystem, injector: I
   }
 
   private val registeredTypeNames = new ConcurrentHashMap[String, Class[_]]()
+  private val reverseRegister: (Class[_]) => Option[String] = { entityClass =>
+    import scala.collection.JavaConversions.mapAsScalaConcurrentMap
+    registeredTypeNames.filter { case (name, className) => className == entityClass }.keys.headOption
+  }
```

Comment:
Use a function to reverse lookup so we only need to keep one copy of the mapping. I avoid using two because there could be a race condition in https://github.com/lagom/lagom/pull/459/files?diff=unified#diff-b00d9e70042b9ecf936fb5cc01bb8d1fR90 If we have to maintain another map. The alternative was created a synchronized block.